src/main/java/com/pusher/client/Pusher.java

```diff
@@ -105,15 +101,15 @@ public Pusher(final String apiKey, final PusherOptions pusherOptions) {
 
         this.pusherOptions = pusherOptions;
         this.factory = factory;
-        connection = factory.getConnection(apiKey, this.pusherOptions, this::handleEvent);
+        connection = factory.getConnection(apiKey, this.pusherOptions, this);
         channelManager = factory.getChannelManager();
         user = factory.newUser(connection, pusherOptions.getUserAuthenticator());
         channelManager.setConnection(connection);
     }
 
-    private void handleEvent(String event, String wholeMessage) {
-        user.handleEvent(event, wholeMessage);
-        channelManager.onMessage(event, wholeMessage);
+    public void handleEvent(PusherEvent event) {
+        user.handleEvent(event);
+        channelManager.handleEvent(event);
     }
```

Comment:
This should remain private. I don't think there's a benefit to exposing it.

Reply:
It's implementing the PusherEventHandler interface now, so it has to be public

Reply:
But what's the benefit of implementing that interface if we don't want to expose this method?

Reply:
Previously it used a BiConsumer, which is kind of magic. Now it's a nice interface implementation, passed to factory.getConnection

Reply:
On the call to `getConnection`, we could keep it as it was with `this::handleEvent` then `Pusher` doesn't have to implement anything if you make `getConnection` take a `Consumer<PusherEvent>` or if you annotate `PusherEventHandler` as a `FunctionalInterface`

Reply:
ok, like this?